### DISCUSSION TITLE
Sensor not updating to 2.3.10

### DISCUSSION THREAD

**Discussion by arjoshi89** [Security-Onion-Solutions/securityonion#2056] [Unsupported Versions] [answered]
Distributed Environment: manager, sensor1 (Heavy Node), sensor2 (Heavy node)
- Settings: Each node connects to the Internet for updates
Confirming that every node has internet connection

### **Issue: Sensors are still showing old version after update**

Used sudo soup command on manager to update. 
***********************************
Manager gets updated as below:
[root@securityonion ~]# cat /etc/soversion
2.3.10
*******************************
But both sensors are still showing old version after update:
[root@sensor2 ~]# cat /etc/soversion
2.3.1

[root@sensor1 ~]# cat /etc/soversion
2.3.1

************************

[root@securityonion SecurityOnion]# salt '*' test.ping
securityonion_manager:
    True
sensor1_heavynode:
    True
sensor2_heavynode:
    True
[root@securityonion SecurityOnion]#

**************************
Checked soup.log from manager, it looks overwritten

[root@securityonion ~]# cat soup.log
Usage: cmd [-b]
Checking to see if this is a manager.

This is a manager, We can proceed.
Checking to see if this is an airgap install

Found that Security Onion 2.3.10 is currently installed.

Cloning Security Onion github repo into /tmp/sogh/securityonion.
Cloning into 'securityonion'...

Verifying we have the latest soup script.
This version of the soup script is up to date. Proceeding.

Let's see if we need to update Security Onion.
You are already running the latest version of Security Onion.
[root@securityonion ~]#
*****************************************************
[root@securityonion ~]# sudo salt-key -l all
Accepted Keys:
securityonion_manager
sensor1_heavynode
sensor2_heavynode
Denied Keys:
Unaccepted Keys:
Rejected Keys:
****************************************************
[root@securityonion ~]#  sudo salt '*' test.version
sensor1_heavynode:
    3001.1
sensor2_heavynode:
    3002.2
securityonion_manager:
    3002.2
***********************************
- Strange but UI only shows sensor2 now, it does not display sensor1. All the services under sensor1 showed as Error. 

Please help me with this issue.

**Comment by weslambert** [comment]
It may take some time before the changes are propagated.  Is this still an issue?

**Reply by arjoshi89** [reply]
Yes, I tried it but still sensor showing old version. It is still issue.

**Comment by TOoSmOotH** [comment, marked as answer]
The issue is it doesn't update /etc/soversion on the non managers. sensor1 looks to be the only one who hasn't updated yet. You can connect to it and do a sudo salt-call state.highstate on it to force it to update. There are changes coming in the next release where you will see the version change in that file on non managers.

**Reply by arjoshi89** [reply]
I performed the step but still sensor1 not showing under UI:

[root@sensor1 ~]# sudo salt-call state.highstate
local:
----------
          ID: minion-state-apply-test
    Function: file.touch
        Name: /opt/so/log/salt/state-apply-test
      Result: True
     Comment: Updated times on file /opt/so/log/salt/state-apply-test
     Started: 14:08:20.643660
    Duration: 9.827 ms
     Changes:
              ----------
              touched:
                  /opt/so/log/salt/state-apply-test
----------
          ID: salt_bootstrap
    Function: file.managed
        Name: /usr/sbin/bootstrap-salt.sh
      Result: True
     Comment: File /usr/sbin/bootstrap-salt.sh is in the correct state
     Started: 14:08:20.653667
    Duration: 40.596 ms
     Changes:
----------
          ID: install_salt_minion
    Function: cmd.run
        Name: exec 0>&- # close stdin
exec 1>&- # close stdout
exec 2>&- # close stderr
nohup /bin/sh -c 'yum versionlock delete "salt-*" && /usr/sbin/bootstrap-salt.sh -X -s 120 -F -x python3 stable 3002.2 && yum versionlock add "salt-*"' &

      Result: True
     Comment: Command "exec 0>&- # close stdin
              exec 1>&- # close stdout
              exec 2>&- # close stderr
              nohup /bin/sh -c 'yum versionlock delete "salt-*" && /usr/sbin/bootstrap-salt.sh -X -s 120 -F -x python3 stable 3002.2 && yum versionlock add "salt-*"' &
              " run
     Started: 14:08:20.695079
    Duration: 2256.306 ms
     Changes:
              ----------
              pid:
                  1381
              retcode:
                  0
              stderr:
              stdout:
----------
          ID: salt_minion_package
    Function: pkg.installed
      Result: True
     Comment: onlyif condition is false
     Started: 14:08:22.981221
    Duration: 28.694 ms
     Changes:
----------
          ID: salt_minion_service
    Function: service.running
        Name: salt-minion
      Result: True
     Comment: onlyif condition is false
     Started: 14:08:23.010335
    Duration: 28.671 ms
     Changes:

Summary for local
------------
Succeeded: 5 (changed=2)
Failed:    0
------------
Total states run:     5
Total run time:   2.364 s
[root@sensor1 ~]#




***********************************

[root@securityonion ~]# sudo salt '*' test.version
sensor2_heavynode:
    3002.2
sensor1_heavynode:
    3001.1
securityonion_manager:
    3002.2
[root@securityonion ~]#
*************************************

**Reply by TOoSmOotH** [reply]
Are you using Ubuntu? If so run it again and it will show up.

**Reply by arjoshi89** [reply]
I run the command multiple times but still same. It is installed directly on hardware.

**Reply by m0duspwnens** [reply]
@arjoshi89 
On sensor1, if you run `rpm -qa|grep salt` does it still show that only version 3001.1? If so, is there anything in the log at `/tmp/bootstrap-salt.log` showing that the new package is failing to install?

**Reply by arjoshi89** [reply]
Yes, it still shows 3001.1 on sensor1

[root@sensor1 ~]# rpm -qa|grep salt
salt-3001.1-1.el7.noarch
salt-minion-3001.1-1.el7.noarch
salt-master-3001.1-1.el7.noarch

[root@sensor1 tmp]# cat bootstrap-salt.log
 *  INFO: Running version: 2020.06.23
 *  INFO: Executed by:
 *  INFO: Command line: '/usr/sbin/bootstrap-salt.sh -X -s 120 -F -x python3 stable 3002.2'
 *  INFO: Detected -x option. Using python3 to install Salt.

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   3.10.0-1127.19.1.el7.x86_64
 *  INFO:   Distribution: CentOS 7.8

 *  INFO: Installing minion
 *  INFO: Daemons will not be started
 *  INFO: Found function install_centos_stable_deps
 *  INFO: Found function config_salt
 *  INFO: Found function preseed_master
 *  INFO: Found function install_centos_stable
 *  INFO: Found function install_centos_stable_post
 *  INFO: Found function install_centos_restart_daemons
 *  INFO: Found function daemons_running
 *  INFO: Found function install_centos_check_services
 *  INFO: Running install_centos_stable_deps()
 * ERROR: https://repo.saltstack.com/py3/redhat/7/x86_64/archive/3002.2/SALTSTACK-GPG-KEY.pub failed to download to /tmp/salt-gpg-uIOBYFDL.pub
 * ERROR: Failed to run install_centos_stable_deps()!!!
[root@sensor1 tmp]# ^C

**Reply by m0duspwnens** [reply]
It looks like it is failing due to the error above. Are you able to test the connection by using `wget` to get that file? Is `/tmp` full by any chance?

**Reply by arjoshi89** [reply]
Thank you. /tmp is not full.

But, repo.saltstack.com not resolving. 
[root@sensor1 tmp]# wget https://repo.saltstack.com/py3/redhat/7/x86_64/archive/3002.2/SALTSTACK-GPG-KEY.pub
--2020-11-24 15:52:40--  https://repo.saltstack.com/py3/redhat/7/x86_64/archive/3002.2/SALTSTACK-GPG-KEY.pub
Resolving repo.saltstack.com (repo.saltstack.com)... failed: Name or service not known.
wget: unable to resolve host address ‘repo.saltstack.com’

Where can i check/change DNS settings in sensor? And also IP settings

**Reply by m0duspwnens** [reply]
DNS settings are in `/etc/resolv.conf` and network settings will be in `/etc/sysconfig/network-scripts/ifcfg-<interface_name>`.

**Reply by arjoshi89** [reply]
Thank you all for providing hint, Corrected it. Now I can reach to the site. 

I run below command under sensor to update salt:
sudo yum update salt

Now, sensor and manager have same salt version. I can see sensor1 under UI now.
